metron-platform/metron-enrichment/metron-enrichment-common/src/test/java/org/apache/metron/enrichment/cache/ObjectCacheTest.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.metron.enrichment.cache;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.metron.common.utils.SerDeUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ObjectCacheTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  FileSystem fs;
+  List<String> data;
+  ObjectCache cache;
+
+  @Before
+  public void setup() throws IOException {
+    fs = FileSystem.get(new Configuration());
+    data = new ArrayList<>();
+    {
+      data.add("apache");
+      data.add("metron");
+      data.add("is");
+      data.add("great");
+    }
+    cache = new ObjectCache();
+  }
+
+  @Test
+  public void test() throws Exception {
+    String filename = "target/ogt/test.ser";
+    Assert.assertTrue(cache.getCache() == null || !cache.getCache().asMap().containsKey(filename));
+    assertDataIsReadCorrectly(filename);
+  }
+
+  public void assertDataIsReadCorrectly(String filename) throws IOException {
+    try(BufferedOutputStream bos = new BufferedOutputStream(fs.create(new Path(filename), true))) {
+      IOUtils.write(SerDeUtils.toBytes(data), bos);
+    }
+    cache.initialize(ObjectCacheConfig.fromGlobalConfig(new HashMap<>()));
```

Comment:
Initializing the cache requires some `ObjectCacheConfig` properties to be set, otherwise you get NPEs.  I just did this for convenience.  I'm happy to change it to `new ObjectCacheConfig()` and explicitly set the cache properties.

Reply:
It sounds like you might mean to have the no-arg default constructor set to `private ObjectCacheConfig() {}` because using it causes unrecoverable errors. Alternatively, you could also do away with `fromGlobalConfig` and simply create
```
public ObjectCacheConfig(Map<String, Object> globalConfig) {
...
}
```
but I don't think we should have both. I personally prefer the latter approach and leveraging constructors for their intended purpose. Let's let the API dictate the use rather than relying on javadoc or, worse yet, NPEs only found at runtime. 

Side note - what specifically causes NPEs? Can we address that as well, while we're at it?

Reply:
You're right.  The no arg constructor doesn't really serve any purpose.  I will change `fromGlobalConfig` to a constructor (should have done this in the first place).

The NPE comes from Caffeine code.  As long as we're not passing in null values to the cache builder it's not an issue.

Reply:
See my other recent comments on this - can we setup some sensible defaults for the config? I think the pattern we've attempted to follow is to maximize configurability of the system by exposing as many options as possible, but also making easy to get up and running with some defaults that they can configure PRN. I'd look at the cache used in `ParallelEnricher` to see what has been used for defaults in other caches.

Reply:
I believe moving from `fromGlobalConfig` to a constructor will solve this since that method sets sensible defaults.

Reply:
Done with the latest commit.